mern/client/src/Components/Modals/GenericModal.js

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import "./ModalStyling.css";
+import GenericButton from "../Buttons/GenericButton";
+
+/**
+ * A generic input modal component
+ * @module GenericModal
+ */
+/** @prop {boolean} [show] : whether the modal shows or not*/
```

Comment:
question: what do these prop decorators do? is this just for documentation sake?

Reply:
Yes. Documentation. If you hover over the Component tag it'll show you a description and the props (i.e. parameters that can go into it and what they do)